src/main/scala/com/mozilla/telemetry/mozdata/Hadoop.scala

```diff
@@ -0,0 +1,76 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package com.mozilla.telemetry.mozdata
+
+import org.apache.commons.io.IOUtils
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.{FileSystem, Path}
+
+object Hadoop {
+  /** List the directories and files within a directory using hadoop
+    *
+    * @param directory path to the directory to list
+    * @return (dirnames, filenames) found in the directory. omits symlinks.
+    */
+  def ls(directory: String): (List[String], List[String]) = {
```

Comment:
It doesn't look like this function differentiates between directories that don't exist, and those that do but are empty. It probably makes sense to raise an error for an ls on a non-existent path.